sqrl-planner/src/main/java/com/datasqrl/plan/rules/AnnotatedLP.java

```diff
@@ -254,6 +241,19 @@ public AnnotatedLP inlineTopN(RelBuilder relBuilder, ExecutionAnalysis exec) {
       }
       return AnnotatedLP.build(relBuilder.build(), type, newPk, timestamp, newSelect, this)
           .sort(newSort).build();
+    } else { //!distinct && (!hasPartition || !hasLimit)
```

Comment:
leave the else if(s) and throw on else that doesn't match?

Reply:
I just added the negation of the if clause here to make to remind myself of what this is. There is no other else after that.